src/librustc_lexer/src/lib.rs

```diff
@@ -179,21 +179,19 @@ pub enum Base {
 /// but shebang isn't a part of rust syntax.
 pub fn strip_shebang(input: &str) -> Option<usize> {
     // Shebang must start with `#!` literally, without any preceding whitespace.
+    // For simplicity we consider any line starting with `#!` a shebang,
+    // regardless of restrictions put on shebangs by specific platforms.
     if input.starts_with("#!") {
+        // Ok, this is a shebang but if the next non-whitespace token is `[` or maybe
+        // a doc comment (due to `TokenKind::(Line,Block)Comment` ambiguity at lexer level),
+        // then it may be valid Rust code, so consider it Rust code.
         let input_tail = &input[2..];
```

Comment:
Did you forget to push the change?

Reply:
No, I was waiting for some response from the pinged people or someone else from the compiler or lang team.

I guess enough time has passed, I'll update the PR with an aim of landing it and fixing https://github.com/rust-lang/rust/issues/73250 instead of considering it wontfix.